RevenueCatUI/CustomerCenter/ViewModels/CustomerCenterViewModel.swift

```diff
@@ -131,7 +152,21 @@ import RevenueCat
             return .purchasesNotFound
         }
     }
+}
 
+fileprivate extension String {
+    /// Takes the first characters of this string, if they conform to Major.Minor.Patch. Returns nil otherwise.
+    /// Note that Minor and Patch are optional. So if this string starts with a single number, that number is returned.
+    func takeVersion() -> String? {
```

Comment:
Maybe this function could be called something like `versionString()`? `takeVersion` doesn't read for me. Could also consider making this a computed property

```swift
var versionString: String? {
...
}

Reply:
Right, I can imagine that. `takeVersion()` was supposed to be analogous to Kotlin's use of the terms [_take_ and _drop_](https://kotlinlang.org/docs/collection-parts.html#take-and-drop) which apply to collections and strings.

`String.take(Int)` takes the first `n` characters from a string. We're kinda doing the same, but instead of `n` we take the first characters conforming to a version string.

I renamed it to `versionString()`, but kept it as a function. I felt that a function better signifies that we're doing some work on the string, and `versionString` is not really a property of a string. Weakly held opinion though.

Reply:
Sounds good!